src/replaying.jl

```diff
@@ -61,3 +61,36 @@ function avoid_allocations(record, f, args...; kwargs...)
     ctx = new_replay_ctx(record)
     return Cassette.overdub(ctx, f, args...; kwargs...)
 end
+
+struct FrozenFunction{F}
+    f::F
+    ctx::Dict{Tuple, ReplayCtx}
+    FrozenFunction(f) = new{typeof(f)}(f, Dict{Tuple, ReplayCtx}())
+end
+
+@generated function (f::FrozenFunction)(xs...)
+    return quote
+        if haskey(f.ctx, $xs)
+            ctx = f.ctx[$xs]
+            # step = ctx.metadata.step::Ref{Int}
+            ctx.metadata.step[] = 1
```

Comment:
Why not using the function to reset a replay?

Reply:
err, which function?

Reply:
`reinitialize!`

Reply:
```julia
function reinitialize!(record)
    for ii in eachindex(record.allocations)
        alloc = record.allocations[ii]
        sz = record.initial_sizes[ii]
        if size(alloc) !== sz
            # fix any vectors that were e.g. `push!`ed to.
            resize!(alloc, sz...)
        end
    end
    return record
end
```

you mean this? This won't reset the `step`  IIUC.

Reply:
Correct, because it resets a record, and then a `ReplayCtx` is created based on the record, which will have initial step of 0.

Reply:
but I think my point is the current implementation will get you rid of the allocation of `Ref`, so it will result in perfect  0 allocations as long as there is no type infer problem.